pyanaconda/modules/payloads/source/cdn/cdn.py

```diff
@@ -44,4 +53,15 @@ def type(self):
     @property
     def description(self):
         """Get description of this source."""
-        return _("Red Hat CDN")
+
+        registered_to_satellite = False
+        if self._subscription_module:
+            registered_to_satellite = self._subscription_module.IsRegisteredToSatellite
+
+        # if we are currently connected to a Satelite instance, call the source "Satellite"
+        # to make it clear a repo provided by the Satellite instance is used, not the global
+        # Red Hat CDN
+        if registered_to_satellite:
+            return _("Satellite")
+        else:
+            return _("Red Hat CDN")
```

Comment:
Isn't there a general way to describe both sources since the Payloads module doesn't care at all? This is not a good idea. If you really want to have a full control over the description, we can implement a setter you can call. However, I think that the Installation Source spoke doesn't have to provide this information.

Reply:
We really need to be specific there - customers using the the Red Hat hosted infrastructure need to see they are indeed using our CDN as the installation source and are getting all the related benefits (global availability, up to date packages, etc.).

On the other hand customers with a Satellite instance very explicitly choose to use their self hosted subscription infrastructure so seeing "Red Hat CDN" as the installation source would be very confusing & could generate significant support burden as it would appear as something went wrong and the Satellite hosted repositories are not in use.

Also, some sources already have dynamic decription (NFS, harddrive), is it that different here ?

Reply:
The dynamic descriptions are not a problem, but you shouldn't do DBus calls from the main threads of the DBus modules. See my answer above.

Reply:
The dynamic logic has been dropped.